server/src/main/java/org/elasticsearch/script/StringFieldScript.java

```diff
@@ -43,6 +43,23 @@ public boolean isResultDeterministic() {
         }
     };
 
+    public static final StringFieldScript.Factory PARSE_FROM_SOURCE_PATHS = new Factory() {
```

Comment:
It's probably worth javadoc to explain how this is different from `PARSE_FROM_SOURCE`.

Reply:
The runtime fields change is definitely the part I was least confident in the solution for. Should source paths be used for all source-only runtime fields? Would that make more sense than having two of these? Does it ever make sense to specify a source-only field as part of the runtime mappings where the source would be part of the parent field?

Reply:
> Should source paths be used for all source-only runtime fields?

Truly runtime fields *can't* have sub-fields. That concept only exists in the mapping and runtime fields just have a single path.

Reply:
That isn't a simple thing, for what it's worth. We could have built sub-fields into runtime fields. We just didn't because we didn't know why we would have. We didn't think, at the time, that they'd *do* anything.